src/test/java/core/basesyntax/service/operation/PurchaseHandlerImplTest.java

```diff
@@ -0,0 +1,56 @@
+package core.basesyntax.service.operation;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitRecord;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class PurchaseHandlerImplTest {
+    private static OperationHandler operationHandler;
+    private Map<String, Integer> expected;
+    private FruitRecord fruitRecord;
+
+    @BeforeAll
+    static void setUp() {
+        operationHandler = new PurchaseHandlerImpl();
+    }
+
+    @Test
+    void checkFruitPurchase_Ok() {
+        expected = new HashMap<>();
+        expected.put("banana",10);
+        Storage.fruitMap.put("banana",20);

```

Comment:
For what is this line here?

Reply:
I create the expected MAP and then add 20 banana into storage directly.  After that I subtract 10 bananas from  Storage using purchase operation. Finally compare actual map and expected

Reply:
do we need to clear the storage in this test? Usually each test that contains storage integration should clean the storage after all, and even better, after each case

Reply:
done. created @After for all the tests with storage use